src/code/z_rumble.c

```diff
@@ -0,0 +1,85 @@
+#include "global.h"
+#include "z64rumble.h"
+
+RumbleManager sRumbleMgr;
+
+void Rumble_Update(void* arg0) {
+    RumbleManager_Update(&sRumbleMgr);
+    PadMgr_RumbleSet(sRumbleMgr.rumbleEnabled);
+}
+
+// Used by some bosses (and fishing)
+void func_8013EC44(f32 distSq, u8 arg1, u8 arg2, u8 arg3) {
+    s32 temp;
+    s32 distance;
+
+    if (SQ(1000.0f) < distSq) {
+        distance = 1000;
+    } else {
+        distance = sqrtf(distSq);
+    }
+
+    if ((distance < 1000) && (arg1 != 0) && (arg3 != 0)) {
+        temp = arg1 - (distance * 255) / 1000;
+
+        if (temp > 0) {
+            sRumbleMgr.unk_10A = temp;
+            sRumbleMgr.unk_10B = arg2;
+            sRumbleMgr.unk_10C = arg3;
+        }
+    }
+}
+
+// Normal rumble?
+void func_8013ECE0(f32 xyzDistToPlayerSq, u8 arg1, u8 arg2, u8 arg3) {
```

Comment:
```suggestion
void func_8013ECE0(f32 distSq, u8 arg1, u8 arg2, u8 arg3) {
```
probably better since I don't think the function is specific to player distance, also would mirror `func_8013EC44` better.
Note this would also need to be updated in the `.h` file.